backend/src/authentication/PlaintextAuthenticationProvider.ts

```diff
@@ -9,8 +9,24 @@ export default class PlaintextAuthenticationProvider
     username: string,
     password: string
   ): Promise<AuthenticationResult> {
-    if (password === "p4") {
-      return { username: username, userid: username, type: "plain" };
+    if (process.env.BACKEND_USERS != undefined) {
```

Comment:
I suggest we create a new AuthenticationProvider here for the time being so we can handle both use cases easily.

Reply:
I also added a constructor to the new provider to check existance of the required env var